sys/include/congure/quic.h

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_congure_quic    CongURE implementation of QUIC's CC
+ * @ingroup     sys_congure
+ * @brief       Implementation of QUIC's congestion control algorithm for the
+ *              CongURE framework.
+ * @{
+ *
+ * @file
+ *
+ * @author  Martine S. Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef CONGURE_QUIC_H
+#define CONGURE_QUIC_H
+
+#include "ztimer.h"
+
+#include "congure.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Constants for the congestion control.
+ *
+ * Example usage:
+ *
+ * ~~~~~~~~~~~~~~~~ {.c}
+ * static const congure_quic_snd_consts_t consts = {
+ *     .cong_event_cb = _maybe_send_one_pkt,
+ *     .init_wnd = 12000,
+ *     .min_wnd = 2400,
+ *     .init_rtt = 333,
+ *     .max_msg_size = 1200,
+ *     .pc_thresh = 3000,
+ *     .granularity = 1,
+ *     .loss_reduction_numerator = 1,
+ *     .loss_reduction_denominator = 2,
+ *     .inter_msg_interval_numerator = 5,
+ *     .inter_msg_interval_denominator = 4,
+ * }
+ * static congure_quic_snd_t cong;
+ *
+ * // ...
+ * congure_quic_snd_setup(&cong, &const);
+ * ~~~~~~~~~~~~~~~~
+ */
+typedef struct {
+    /**
+     * @brief   congestion event callback
+     *
+     * This callback is called when congestion event is detected by
+     * message loss or a CE notification. QUIC typically uses this to send
+     * a packet to speed up loss
+     * recovery.
+     *
+     * @param[in] ctx   callback context
+     */
+    void (*cong_event_cb)(void *ctx);
+
+    /**
+     * @brief   Initial congestion window size in initiator-defined units.
```

Comment:
What are initiator-defined units?

Reply:
The units that are defined by the programmer at initialization of the object. Like all window units in CongURE they are arbitrary (could be bytes as in TCP or fragments as in 6LoWPAN), but consistent within one CongURE instance.

Reply:
(i.e. if you are at this point of reading this, you hopefully know how CongURE handles units so it should be self-explanatory).